### PR TITLE
Wurst spears

### PR DESCRIPTION
**Pull request by Marsunpaisti**

### CODE REVIEW

File: wurst/_items/Spears.wurst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        case PoisonSpear
+            spawnItemChance(0.5, ITEM_POISON_SPEAR, target.getPos())
+        case DarkSpear
+            spawnItemChance(0.4, ITEM_DARK_SPEAR, target.getPos())
+            let manaBurnPerTick = GetRandomReal(45, 60) / 4
+            ManaBurn(target, manaBurnPerTick)
+            doPeriodicallyCounted(1.00, 3) manaBurnLoop ->
+                ManaBurn(target, manaBurnPerTick)
+        case SteelSpear
+            spawnItemChance(0.6, ITEM_STEEL_SPEAR, target.getPos())
```
Comment (jonathanmclaus) [inline wurst/_items/Spears.wurst]: Can you change these back to 0.66?

File: wurst/_items/Spears.wurst [hunk 2]
```diff
@@ -0,0 +1,99 @@
+package Spears
+import ClosureEvents
+import ITTDummyCaster
```
[the remark this answers is not in the document]
Comment (Marsunpaisti) [reply wurst/_items/Spears.wurst]: The unit associated with the dummycaster instance cant be accessed to apply the listener. I put up a PR to stdlib2 already

File: wurst/_items/Spears.wurst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    switch spearType
+        case UltraPoisonSpear
+            spawnItemChance(0.5, ITEM_ULTRA_POISON_SPEAR, target.getPos())
+        case RefinedPoisonSpear
+            spawnItemChance(0.5, ITEM_REFINED_POISON_SPEAR, target.getPos())
+        case PoisonSpear
+            spawnItemChance(0.5, ITEM_POISON_SPEAR, target.getPos())
+        case DarkSpear
+            spawnItemChance(0.4, ITEM_DARK_SPEAR, target.getPos())
+            let manaBurnPerTick = GetRandomReal(45, 60) / 4
```
[the remark this answers is not in the document]
Comment (Marsunpaisti) [reply wurst/_items/Spears.wurst]: Almost unnoticeable. Mostly to further increase consistency

File: wurst/_items/Spears.wurst (hunk 1 above)
Comment (Marsunpaisti) [reply wurst/_items/Spears.wurst]: My mistake for some reason I converted 2/3 -> 0.6

Comment (Marsunpaisti) [reply wurst/_items/Spears.wurst]: Updated

File: wurst/_items/Spears.wurst (hunk 3 above)
[the remark this answers is not in the document]
Comment (Marsunpaisti) [reply wurst/_items/Spears.wurst]: Actually how about we just set it to a consistent 50 total with 10 burn per 0.75sec tick

File: wurst/_items/Spears.wurst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        case RefinedPoisonSpear
+            spawnItemChance(0.5, ITEM_REFINED_POISON_SPEAR, target.getPos())
+        case PoisonSpear
+            spawnItemChance(0.5, ITEM_POISON_SPEAR, target.getPos())
+        case DarkSpear
+            spawnItemChance(0.4, ITEM_DARK_SPEAR, target.getPos())
+            let manaBurnPerTick = 10.
+            ManaBurn(target, manaBurnPerTick)
+            doPeriodicallyCounted(0.75, 4) manaBurnLoop ->
+                ManaBurn(target, manaBurnPerTick)
```
Comment (theQuazz) [inline wurst/_items/Spears.wurst]: Hmm just going to flag this with request changes until council decides on this

File: wurst/_items/Spears.wurst (hunk 2 above)
[the remark this answers is not in the document]
Comment (Marsunpaisti) [reply wurst/_items/Spears.wurst]: Thats the plan here

File: wurst/_items/Spears.wurst (hunk 4 above)
Comment (jonathanmclaus) [reply wurst/_items/Spears.wurst]: I think the stdlib2 should have an immediate option for `doPeriodicallyCounted`, so you aren't force to manually call the callback. Thoughts?

**Comment by Marsunpaisti** [comment]
Reverted manaburn over time on darkspears

File: wurst/_items/Spears.wurst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    switch spearType
+        case UltraPoisonSpear
+            spawnItemChance(0.5, ITEM_ULTRA_POISON_SPEAR, target.getPos())
+        case RefinedPoisonSpear
+            spawnItemChance(0.5, ITEM_REFINED_POISON_SPEAR, target.getPos())
+        case PoisonSpear
+            spawnItemChance(0.5, ITEM_POISON_SPEAR, target.getPos())
+        case DarkSpear
+            spawnItemChance(0.4, ITEM_DARK_SPEAR, target.getPos())
+            ManaBurn(target, GetRandomReal(45, 55))
```
Comment (theQuazz) [inline wurst/_items/Spears.wurst]: Even though I said we shouldn't do a drain over time I still think we should do a resistance similar to `ABILITY_SPEAR_RESIST`

Comment (Marsunpaisti) [reply wurst/_items/Spears.wurst]: Added the system

**Pull request merged**